src/main/g8/$name__norm$-impl/src/main/scala/$package$/impl/$name__Camel$Entity.scala

```diff
@@ -29,61 +39,75 @@ import scala.collection.immutable.Seq
+object $name;format="Camel"$Behavior {
 
-  override type Command = $name;format="Camel"$Command[_]
-  override type Event = $name;format="Camel"$Event
-  override type State = $name;format="Camel"$State
 
-  /**
-    * The initial state. This is used if there is no snapshotted state to be found.
-    */
-  override def initialState: $name;format="Camel"$State = $name;format="Camel"$State("Hello", LocalDateTime.now.toString)
-
-  /**
-    * An entity can define different behaviours for different states, so the behaviour
-    * is a function of the current state to a set of actions.
-    */
-  override def behavior: Behavior = {
-    case $name;format="Camel"$State(message, _) => Actions().onCommand[UseGreetingMessage, Done] {
-
-      // Command handler for the UseGreetingMessage command
-      case (UseGreetingMessage(newMessage), ctx, state) =>
-        // In response to this command, we want to first persist it as a
-        // GreetingMessageChanged event
-        ctx.thenPersist(
-          GreetingMessageChanged(newMessage)
-        ) { _ =>
-          // Then once the event is successfully persisted, we respond with done.
-          ctx.reply(Done)
-        }
-
-    }.onReadOnlyCommand[Hello, String] {
+  def create(entityContext: EntityContext[$name;format="Camel"$Command]): Behavior[$name;format="Camel"$Command] = {
+    val persistenceId: PersistenceId = PersistenceId(entityContext.entityTypeKey.name, entityContext.entityId)
 
-      // Command handler for the Hello command
-      case (Hello(name), ctx, state) =>
-        // Reply with a message built from the current message, and the name of
-        // the person we're meant to say hello to.
-        ctx.reply(s"\$message, \$name!")
+    create(persistenceId)
+      .withTagger(
+        // Using Akka Persistence Typed in Lagom requires tagging your events
+        // in Lagom-compatible way so Lagom ReadSideProcessors and TopicProducers
+        // can locate and follow the event streams.
+        AkkaTaggerAdapter.fromLagom(entityContext, $name;format="Camel"$Event.Tag)
+      )
 
-    }.onEvent {
-
-      // Event handler for the GreetingMessageChanged event
-      case (GreetingMessageChanged(newMessage), state) =>
-        // We simply update the current state to use the greeting message from
-        // the event.
-        $name;format="Camel"$State(newMessage, LocalDateTime.now().toString)
-
-    }
   }
+  private[impl] def create(persistenceId: PersistenceId) = EventSourcedBehavior
+      .withEnforcedReplies[$name;format="Camel"$Command, $name;format="Camel"$Event, $name;format="Camel"$State](
+        persistenceId = persistenceId,
+        emptyState = $name;format="Camel"$State.initial,
+        commandHandler = (cart, cmd) => cart.applyCommand(cmd),
+        eventHandler = (cart, evt) => cart.applyEvent(evt)
+      )
 }
 
 /**
-  * The current state held by the persistent entity.
+  * The current state of the Persistent Entity.
```

Comment:
Maybe we should avoid using the term Persistence Entity when using Akka Persistence Typed.

Some suggestions: 

* Aggregate, because that's how the rest of the world call it*.
* Event Sourced Actor - because `EventSourcedBehavior`
* Event Sourced Model

The problem with `EventSourcedBehavior` is that it removes the notion of modelling. What is  a Behavior? It's the model behavior for sure, in this case the command handlers and event handlers. 

* that's also how we are calling it on the current Akka Persistence Typed docs in Lagom. WIP though.

Reply:
Ah, sure! Better converge on the naming of things early on.

Reply:
Hmmm, should we rename the file to `SomethingAggregate.scala` (and the test to `SmthingAggregateSpec.scala) then?

Reply:
As I'm working on this rename I'm liking the change more and more. Now _entity_ is always used in the context of cluster sharding and _aggregate_ is reserved or the, well, you know, the _aggregate_.